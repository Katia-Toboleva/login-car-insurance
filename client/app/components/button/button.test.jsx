import React from 'react';
import { shallow } from 'enzyme';
import Button from './button';

const getUnit = (props) => (
  shallow(<Button {...props} />)
);

describe('Button', () => {
  describe('render', () => {
    it('should render correctly', () => {
      const wrapper = getUnit();

      expect(wrapper).toMatchSnapshot();
    });

    describe('handleMouseEnter', () => {
      it('updates html correctly', () => {
        const wrapper = getUnit();
        wrapper.find('.button').simulate('mouseEnter');

        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('handleMouseLeave', () => {
      it('updates html correctly', () => {
        const wrapper = getUnit();
        wrapper.find('.button').simulate('mouseLeave');

        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('handleButtonClick', () => {
      it('calls props.onClick', () => {
        const cb = jest.fn();
        const type = 'login';
        const wrapper = getUnit({ onClick: cb, type });
        wrapper.find('.button').simulate('click');

        expect(cb).toHaveBeenCalledWith(type);
      });
    });
  });
});
