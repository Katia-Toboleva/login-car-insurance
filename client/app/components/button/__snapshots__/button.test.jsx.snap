// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render handleMouseEnter updates html correctly 1`] = `
<Button
  active={false}
  onClick={[Function]}
  size="medium"
  text="sign in"
  type="login"
>
  <div
    className="button button--type-login button--hovered button--disabled"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Text
      center={false}
      color="white"
      size="medium"
      text="sign in"
      transform="uppercase"
    >
      <div
        className="text text--size-medium text--color-white text--transform-uppercase"
      >
        <span>
          sign in
        </span>
      </div>
    </Text>
  </div>
</Button>
`;

exports[`Button render handleMouseLeave updates html correctly 1`] = `
<Button
  active={false}
  onClick={[Function]}
  size="medium"
  text="sign in"
  type="login"
>
  <div
    className="button button--type-login button--disabled"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Text
      center={false}
      color="white"
      size="medium"
      text="sign in"
      transform="uppercase"
    >
      <div
        className="text text--size-medium text--color-white text--transform-uppercase"
      >
        <span>
          sign in
        </span>
      </div>
    </Text>
  </div>
</Button>
`;

exports[`Button render should render correctly 1`] = `
<Button
  active={false}
  onClick={[Function]}
  size="medium"
  text="sign in"
  type="login"
>
  <div
    className="button button--type-login button--disabled"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Text
      center={false}
      color="white"
      size="medium"
      text="sign in"
      transform="uppercase"
    >
      <div
        className="text text--size-medium text--color-white text--transform-uppercase"
      >
        <span>
          sign in
        </span>
      </div>
    </Text>
  </div>
</Button>
`;
